### ISSUE TITLE
Unshackle batch mode

### ISSUE THREAD

**Issue by S4mpsa** [GTNewHorizons/GT-New-Horizons-Modpack#14002]
### Your GTNH Discord Username

Sampsa_

### Your Pack Version

Nightly

### Your Proposal

Context: Batch mode is a a mode that most machines (ALL machines after GenericProcessingLogic is fully rolled out) have, which tries to multiply inputs, outputs and processing times by up to 128 times to avoid a large amount of recipe checks in a short amount of time. This was primarily added to combat lag, so the minimum processing time was capped at 1 * Batch Multiplier ticks, essentially making sure you cannot make the machine go any faster.

**Proposal**

Remove the minimum tick limit from Batch Mode, allowing overclocks to happen even after a recipe is 1-ticking if batch mode is enabled. This would affect balance, as it would, on average, make machines work faster in lategame.

There are multiple reasons why I, and many other endgame players, think this should be the case:

- Machine spam has gotten out of hand in extreme lategame, which is caused by the amount of items you need being so high, while machines are (now artificially) capped at only being able to process a single recipe in a tick. Removing the limit from batch mode would allow a huge reduction in the amount of machines required, in exchange for increased energy usage.
- Currently most machines being fed UXV power, doing LV to UEV recipes (so like 90% of them), waste upwards of 99% of the energy input in the extreme cases because they cannot overclock past 1 tick. Unshackling batch mode would allow all machines to utilize their entire energy to be utilized.
- This is up to personal opinion, but the 1-tick limit is largely an engine limitation instead of a gameplay mechanic, so being able to circumvent engine limitations now that we have the tools to do so makes sense. With the numbers in lategame GTNH becoming so large, it feels backwards to stay shackled to the 20-ticks-per-second of base Minecraft when we are making our own energy systems to get around the limitations of base Gregtech.

Now, most of the time this has been discussed, the primary counterpoint to unshackling is that it was originally intended as a lag reduction tool, and allowing overclocking past the 128-tick limit would just make machines 1-tick again. What we can do here is allow a dynamic batch mode: If the recipe time drops too low after batch mode, increase the batch multiplier even more until the recipe time is above some threshold.

There are multiple other ways of improving batch mode which I would like to implement, but this change is the only thing affecting balance, and warrants its own discussion.

### Your Goal

Lategame machine spam has gotten out of hand, allowing machines to bypass the 1-tick limit would help.

### Your Vision

Get rid of artificial limits imposed upon us by engine limitations.

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying *I like this change please reconsider adding it* will prompt us to investigate and reopen it once we confirm your report.
- [X] I understand this change request may not attract enough attention and thus not be implemented.
- [X] I understand this change request may be rejected due to other community members think it's inappropriate.
- [X] I believe this feature would make the pack better.

**Comment by chochem** [comment]
hmm, so what's your plan for machines with perfect OC in this? This would just give them massive bonus parallels, right? Like what's the point of an MCR if you have a LCR with unshackled batch mode?

**Comment by fluffle** [comment]
Perhaps preventing the LCR from using multi-amp hatches, and allowing the Mega CR to do even more, even faster that way would provide some incentive still? This comes down to trading off greater power consumption for faster processing, so if you want to differentiate LCR from MCR then allowing MCR to consume more power does that in a way that is orthogonal to batching limits.

**Comment by S4mpsa** [comment]
LCR cannot use multi-amp hatches in the first place.

**Comment by Wargenflorgen** [comment]
Strongly support this. Small recipe sizes causes many 1 tick recipes to be tremendous pains in lategame, and keeps most megas from fulfilling their purpose of reducing machine spam. Notable examples include mega EBF/VF, where every recipe in the game 1 ticks at a very low amount of power and the only way to speed them up is to spam more, and compact fusion, where compact mk5 tries to reduce fusion spam but wastes all its OC potential when everything already 1 ticks.

If it's considered a balance issue on certain multis e.g. LCR then maybe they can just not receive batch mode at all, since they currently don't have it -- although LCR would not be viable in lategame even with batch mode, since it can't use TT hatches.

**Comment by Connor-Colenso** [comment]
Honestly this whole batch mode thing just feels like power creep, these powers should belong to a new higher tier variant of certain multiblocks. Not just slapped on every machine. It impacts balance too much.

**Comment by Connor-Colenso** [comment]
Yes but I don't think slapping bandaid fix on it helps, because we all know how it goes. Temporary -> Permanent. This basically just limits you by EU, of which people have so much I don't even know how to phrase it other than in standard form.

**Comment by fluffle** [comment]
I think this comes down to: should players be able to infinitely trade off increased energy inputs for increased processing speed, without some sort of cost?

At the moment, they can achieve this with the cost in player time, server tick time, and resources of building additional machines to run recipes in parallel. Unlocking batch mode removes these costs, so what other costs might replace them, if there must be some sort of cost?

Thoughts:

- Perhaps 2x speed for 4x energy is reasonable, and it should only be possible to get perfect overclocks down to 1 tick per recipe?
- Or some other kind of reduction in the EU->speed multiplier, so that eventually the cost-benefit analysis says it's not worth it?
- Perhaps machines could consume a quantity of fluid to overclock past 1 tick per recipe - a bit like the volcanus consumes pyrotheum?
- Perhaps some sort of expensive multiblock tile entity could be added to unlock the capability so that players had to choose which machines were worth upgrading?

**Comment by Hiagg** [comment]
aal oc for sub 1 tick lol

**Comment by minecraft7771** [comment]
> aal oc for sub 1 tick lol

I wish you good luck if you want to add batch mode to AL/aal :trollface:

**Comment by Sopel97** [comment]
There are already some specific recipes that are just batched copies of smaller ones to counter this limitation, so I don't see why this change shouldn't be made.

**Comment by S4mpsa** [comment]
That's actually a good point, let me see about how we could do that.

**Comment by Hikari1nVoid** [comment]
While we can add extra structures to all MuTEs for higher parallels...The core reason for all machines is being spammed is the exponential growing demand of resource in the endgame and a ridiculous 256x parallel even on mEBF&mVF...
